common/network/neighbor.h

```diff
@@ -20,6 +22,11 @@ typedef struct {
   int flagged;
 } neighbor_t;
 
+bool init_neighbor_with_uri(neighbor_t *const neighbor, char const *const uri);
+bool init_neighbor_with_values(neighbor_t *const neighbor,
+                               protocol_type_t const protocol,
+                               char const *const host, uint16_t const port);
+
 int neighbor_send(neighbor_t *const neighbor,
```

Comment:
what's the difference between this & `tcp_neighbor_send`?

Reply:
neighbor_send was the primary entry point and it would have dispatched to the tcp/udp depending on the neighbour type.
Anyway, it was premature, I'll remove them and will deal properly with it in the broadcaster PR.